Add activity execution summaries and API endpoints

sfmskywalker opened the pull request:
Introduced the ActivityExecutionRecordSummary model and related API endpoints to support listing and retrieving activity execution summaries. Updated the workflow instance viewer and remote service to utilize the new summaries and provide detailed activity execution records on demand.

sfmskywalker committed:
Add activity execution summaries and API endpoints

Introduced the ActivityExecutionRecordSummary model and related API endpoints to support listing and retrieving activity execution summaries. Updated the workflow instance viewer and remote service to utilize the new summaries and provide detailed activity execution records on demand.

sfmskywalker commented on src/framework/Elsa.Studio.Core/Elsa.Studio.Core.csproj:
```diff
@@ -33,12 +33,12 @@
-
     <ItemGroup Label="Elsa">
-        <PackageReference Include="Elsa.Api.Client"/>
+        <ProjectReference Include="..\..\..\..\..\elsa-core\3.2.x\src\clients\Elsa.Api.Client\Elsa.Api.Client.csproj"/>
     </ItemGroup>
 
+    <!--    <ItemGroup Label="Elsa">-->
+    <!--        <PackageReference Include="Elsa.Api.Client"/>-->
+    <!--    </ItemGroup>-->
+
```
Will flip this back as soon as the [elsa-core PR](https://github.com/elsa-workflows/elsa-core/pull/6018) has been merged and new packages are available.

raymonddenhaan commented on src/modules/Elsa.Studio.Workflows/Components/WorkflowInstanceViewer/Components/ActivityDetailsTab.razor.cs:
```diff
@@ -24,17 +24,10 @@ public record ActivityExecutionRecordTableRow(int Number, ActivityExecutionRecor
     [Parameter] public JsonObject Activity { get; set; } = default!;
     
     /// The activity execution records.
```
Let's update the description as well

sfmskywalker replied:
I don't follow. What updated description would do you gave in mind, and why?

sfmskywalker replied:
Do you mean that the object is activity execution **summary** and therefore the comments need to reflect that? If yes, then sure.

sfmskywalker replied:
Updated to use the 'summary" wording. Let me know if you had something else in mind.

raymonddenhaan replied:
I meant that the description above still refers to the collection of records: 
/// The activity execution records.

Instead of:
/// The last activity execution record.

sfmskywalker replied:
Ah, yes I get it now,  I was looking in the wrong file. Thanks.

raymonddenhaan commented on src/modules/Elsa.Studio.Workflows/Components/WorkflowInstanceViewer/Components/ActivityExecutionsTab.razor.cs:
```diff
@@ -12,7 +14,7 @@ public partial class ActivityExecutionsTab
     /// Represents a row in the table of activity executions.
     /// <param name="Number">The number of executions.</param>
     /// <param name="ActivityExecution">The activity execution.</param>
```
Let's keep the param up to date. Do we also want to update the name of the parameter?

raymonddenhaan commented on src/modules/Elsa.Studio.Workflows/Components/WorkflowInstanceViewer/Components/ActivityExecutionsTab.razor.cs:
```diff
@@ -67,11 +71,12 @@ private void CreateSelectedItemDataModels(ActivityExecutionRecord? record)
     {
-        SelectedItem = arg.Item.ActivityExecution;
+        var id = arg.Item.ActivityExecution.Id;
+        var fullRecord = await ActivityExecutionService.GetAsync(id);
```
Can't we assign this directly to SelectedItem?

sfmskywalker replied:
Yes

sfmskywalker committed:
Refactor `ActivityExecution` to `ActivityExecutionSummary`

Renamed all instances of `ActivityExecution` to `ActivityExecutionSummary` for better clarity. Updated corresponding parameters, models, and service references in the affected components.

sfmskywalker committed:
Update LastActivityExecution parameter documentation

Clarified that LastActivityExecution represents the latest activity execution record. This helps in understanding that it is used for displaying the last state of the activity.

sfmskywalker committed:
Merge branch 'blueberry' into enh/activity-executions-lazy-load
